cmd/soroban-cli/src/commands/snapshot/create.rs

```diff
@@ -289,7 +289,14 @@ impl Cmd {
                     }
                     let keep = match &key {
                         LedgerKey::Account(k) => current.account_ids.contains(&k.account_id),
-                        LedgerKey::Trustline(k) => current.account_ids.contains(&k.account_id),
+                        LedgerKey::Trustline(k) => {
+                            current.account_ids.contains(&k.account_id) ||
+                            current.account_ids.iter().any(|id| match &k.asset {
+                                TrustLineAsset::CreditAlphanum4(a4) => a4.issuer == *id,
+                                TrustLineAsset::CreditAlphanum12(a12) => a12.issuer == *id,
```

Comment:
Pulling in every trustline for an asset unfortunately doesn't scale well with the current implementation of snapshots, which was why I left it out. If you pass in the address for a very popular asset, the file becomes huge and unusable.

For assets with much less trust lines though I can see this being very useful so maybe we should add it.

I think this use case will be better served by the following issue that hopefully we can implement in the near future:
- https://github.com/stellar/rs-soroban-sdk/issues/1448

Thoguhts @fnando ?